elephant/conversion.py

```diff
@@ -152,3 +155,704 @@ def binarize(spiketrain, sampling_rate=None, t_start=None, t_stop=None,
+    Calculates the number of bins from given parameter.
+
+    Calculates the number of bins :attr:`num_bins` from the three parameter
+    :attr:`t_start`, :attr:`t_stop` and :attr`binsize`.
+
+    Parameters
+    ----------
+    binsize: quantities.Quantity
+        Size of Bins
+    t_start : quantities.Quantity
+        Start time
+    t_stop: quantities.Quantity
+        Stop time
+
+    Returns
+    -------
+    num_bins : int
+       Number of bins  calculated from given parameter.
+
+    Raises
+    ------
+    ValueError :
+        Raised when :attr:`t_stop` is smaller than :attr:`t_start`".
+
+    """
+    if binsize is not None and t_start is not None and t_stop is not None:
+        if t_stop < t_start:
+            raise ValueError("t_stop (%s) is smaller than t_start (%s)"
+                             % (t_stop, t_start))
+        return int(((t_stop - t_start).rescale(
+            binsize.units) / binsize).magnitude)
+
+
+def calc_binsize(num_bins, t_start, t_stop):
+    """
+    Calculates the stop point from given parameter.
+
+    Calculates the size of bins :attr:`binsize` from the three parameter
+    :attr:`num_bins`, :attr:`t_start` and :attr`t_stop`.
+
+    Parameters
+    ----------
+    num_bins: int
+        Number of bins
+    t_start: quantities.Quantity
+        Start time
+    t_stop
+       Stop time
+
+    Returns
+    -------
+    binsize : quantities.Quantity
+        Size of bins calculated from given parameter.
+
+    Raises
+    ------
+    ValueError :
+        Raised when :attr:`t_stop` is smaller than :attr:`t_start`".
+    """
+
+    if num_bins is not None and t_start is not None and t_stop is not None:
+        if t_stop < t_start:
+            raise ValueError("t_stop (%s) is smaller than t_start (%s)"
+                             % (t_stop, t_start))
+        return (t_stop - t_start) / num_bins
+
+
+def set_start_stop_from_input(spiketrains):
+    """
+    Sets the start :attr:`t_start`and stop :attr:`t_stop` point
+    from given input.
+
+    If one neo.SpikeTrain objects is given the start :attr:`t_stop `and stop
+    :attr:`t_stop` of the spike train is returned.
+    Otherwise the aligned times are returned, which are the maximal start point
+    and minimal stop point.
+
+    Parameters
+    ----------
+    spiketrains: neo.SpikeTrain object, list or array of neo.core.SpikeTrain
+                 objects
+        List of neo.core SpikeTrain objects to extract `t_start` and
+        `t_stop` from.
+
+    Returns
+    -------
+    start : quantities.Quantity
+        Start point extracted from input :attr:`spiketrains`
+    stop : quantities.Quantity
+        Stop point extracted from input :attr:`spiketrains`
+    """
+    if isinstance(spiketrains, neo.SpikeTrain):
+        return spiketrains.t_start, spiketrains.t_stop
+    else:
+        start = max([elem.t_start for elem in spiketrains])
+        stop = min([elem.t_stop for elem in spiketrains])
+    return start, stop
+
+
+class Binned:
```

Comment:
I believe `TimeHistogram` is not a good name, firstly, because a binned spike train can also be clipped (i.e., only a 0/1). In that case it is not a histogram anymore. 

Second, it is not a histogram of time, but more importantly specifically of a spike train. Thus the name is not very telling about what the class does.

Third, `TimeHistogram` sounds more like an analysis, but in fact this class provides more of a data representation.

I would suggest `BinnedSpikeTrain` (similar to the suggestion above), and would leave out "Array" since that is redundant information (and not consistent with the *_to_array functions of the class). This name would also be more compatible with the Neo naming scheme.